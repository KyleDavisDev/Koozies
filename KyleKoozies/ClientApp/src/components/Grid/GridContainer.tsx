import * as React from "react";

// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";

const styles = {
    grid: {
        marginRight: "-15px",
        marginLeft: "-15px",
        width: "auto",
    },
};

const useStyles = makeStyles(styles);

export interface IGridContainerProps {
    children: string | JSX.Element | Array<string | JSX.Element>;
    className?: string | null;
    [key: string]: any;
}

const GridContainer: React.FC<IGridContainerProps> = props => {
    const { children, className, ...rest } = props;
    const classes = useStyles();
    return (
        <Grid container {...rest} className={classes.grid + " " + className}>
            {children}
        </Grid>
    );
}

export default GridContainer;