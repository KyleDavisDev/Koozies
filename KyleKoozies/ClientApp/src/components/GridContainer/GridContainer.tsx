import * as React from "react";

// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";

import GridContainerStyles from "./GridContainerStyles"
const useStyles = makeStyles(GridContainerStyles);

export interface IGridContainerProps {
    children: string | JSX.Element | Array<string | JSX.Element>;
    className?: string | null;
    [key: string]: any;
}

const GridContainer: React.FC<IGridContainerProps> = props => {
    const { children, className, ...rest } = props;
    const classes = useStyles();
    return (
        <Grid container {...rest} className={classes.grid + " " + className}>
            {children}
        </Grid>
    );
}

export default GridContainer;