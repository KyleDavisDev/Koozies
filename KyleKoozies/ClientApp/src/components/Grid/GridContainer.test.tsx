import * as React from 'react';
import * as enzyme from 'enzyme';
import casual from 'casual'
import GridContainer, {IGridContainerProps} from './GridContainer';

casual.seed(15);
const fakeJSXElement = (): JSX.Element => {
    return casual.random_element([
        React.createElement("div", { key: casual.uuid }, casual.string),
        React.createElement("span", { key: casual.uuid }, casual.string)
    ]);
};

const fakeGrid = (): IGridContainerProps => ({
    children: fakeJSXElement(), 
    className: casual.sentence
});

describe("<GridContainer />", () => {
    it('renders without crashing', () => {
        const wrapper = enzyme.shallow(<GridContainer>{fakeJSXElement()}</GridContainer>);
        expect(wrapper).toBeTruthy();
    });
    
    it('matches snapshot', () => {
        const wrapper = enzyme.shallow(<GridContainer>{fakeJSXElement()}</GridContainer>);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders expected children', () => {
        const children = fakeGrid().children;
        const wrapper = enzyme.shallow(<GridContainer>{children}</GridContainer>);
        
        expect(wrapper.find("WithStyles(ForwardRef(Grid))").prop("children")).toBeTruthy();
        expect(wrapper.find("WithStyles(ForwardRef(Grid))").prop("children")).toEqual(children)
    });
})