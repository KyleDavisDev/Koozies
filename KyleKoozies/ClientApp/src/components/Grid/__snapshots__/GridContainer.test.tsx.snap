// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContainer /> matches snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-grid-1 undefined"
  container={true}
>
  <div
    key="36d1a0b1-8b20-477b-8c04-b270f5533ca7"
  >
    aliquid molestiae assumenda laboriosam temporibus odit dignissimos
  </div>
</WithStyles(ForwardRef(Grid))>
`;
